File: src/main/java/ar/edu/um/tesoreria/rest/model/Lectivo.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Lectivo extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 5578990846648088363L;

	@Id
	@Column(name = "lec_id")
	private Integer lectivoId;

	@Column(name = "lec_nombre")
	private String nombre = "";

	@Column(name = "lec_inicio")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fechaInicio;

	@Column(name = "lec_fin")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fechaFinal;

}

File: src/main/java/um/tesoreria/core/kotlin/model/Lectivo.kt
package um.tesoreria.core.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import um.tesoreria.core.model.Auditable
import um.tesoreria.core.util.Jsonifier
import java.time.OffsetDateTime

@Entity
@Table
data class Lectivo(

    @Id @Column(name = "lec_id")
    var lectivoId: Int? = null,

    @Column(name = "lec_nombre")
    var nombre: String = "",

    @Column(name = "lec_inicio")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    var fechaInicio: OffsetDateTime? = null,

    @Column(name = "lec_fin")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    var fechaFinal: OffsetDateTime? = null

) : Auditable() {
    fun jsonify(): String {
        return Jsonifier.builder(this).build()
    }
}
